### PR TITLE
Updates to show exception handling and more protocol method examples

### PR DESCRIPTION
**Pull request by alexwolfmsft**
## Summary

Describe your changes here.

Fixes #Issue_Number (if available)

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/azure/sdk/protocol-convenience-methods.md](https://github.com/dotnet/docs/blob/ba76e85967fe62f3c930b0293a897f311cfa5cea/docs/azure/sdk/protocol-convenience-methods.md) | [docs/azure/sdk/protocol-convenience-methods](https://review.learn.microsoft.com/en-us/dotnet/azure/sdk/protocol-convenience-methods?branch=pr-en-us-42723) |

### CODE REVIEW

**Review by scottaddie — CHANGES_REQUESTED** [review]

File: docs/azure/sdk/protocol-convenience-methods.md
Comment (scottaddie) [inline docs/azure/sdk/protocol-convenience-methods.md:145]: Thinking about this some more, it's probably best to only highlight the catch blocks in these 2 code samples. That is, after all, what we want the customer to focus on.

File: docs/azure/sdk/protocol-convenience-methods.md
Comment (scottaddie) [inline docs/azure/sdk/protocol-convenience-methods.md:137]:
```suggestion
When a service call fails, the service client throws an exception that exposes the HTTP status code and the details of the service response, if available. A `System.ClientModel`-dependent library throws a <xref:System.ClientModel.ClientResultException>, while an `Azure.Core`-dependent library throws a <xref:Azure.RequestFailedException>.
```

**Review by scottaddie — APPROVED** [review]
Looks good to me. Let's wait for @annelo-msft to review before merging.

**Review by annelo-msft — APPROVED** [review]

**Pull request merged**
